src/topp/LuciphorAdapter.cpp

```diff
@@ -0,0 +1,611 @@
+    }
+    
+    vector<String> dcy_neutral_losses = getStringList_("decoy_neutral_losses");
+    for (vector<String>::const_iterator it = dcy_neutral_losses.begin(); it != dcy_neutral_losses.end(); ++it)
+    {
+      config_map["DECOY_NL"].push_back(*it);
+    }
+    
+    return EXECUTION_OK;
+  }
+  
+  void writeConfigurationFile_(const String& out_path, map<String, vector<String> >& config_map)
+  {
+    ofstream output(out_path.c_str());
+    output << "## Input file for Luciphor2 (aka: LucXor). (part of OpenMS)\n\n";
+
+    for (std::map<String, vector<String> >::iterator it = config_map.begin(); it != config_map.end(); ++it)
+    {
+      String key = it->first;
+      if (!key.empty())
+      {
+        for (vector<String>::iterator it_val = it->second.begin(); it_val != it->second.end(); ++it_val)
+        {
+          output << key << " = " << *it_val << "\n";
+        }        
+      }
+    }
+
+    //------------------------------------------------------------------
+    // static parameter definition
+    //------------------------------------------------------------------
+    // Generate a tab-delimited file of all the matched peaks
+    output << "WRITE_MATCHED_PEAKS_FILE = 0\n";
+    
+    output << "MOD_PEP_REP = 0 ## 0 = show single character modifications, 1 = show TPP-formatted modifications\n";
+    
+    output << "## This option can be used to help diagnose problems with Luciphor. Multi-threading is disabled in debug mode.\n";
+    output << "DEBUG_MODE = 0 ## 0 = default: turn off debugging\n";
+    output << "               ## 1 = write peaks selected for modeling to disk\n";
+    output << "               ## 2 = write the scores of all permutations for each PSM to disk\n";
+    output << "               ## 3 = write the matched peaks for the top scoring permutation to disk\n";
+    output << "               ## 4 = write HCD non-parametric models to disk (HCD-mode only option)\n";
+  }
+  
+  void removeTempDir_(const String& temp_dir)
+  {
+    if (temp_dir.empty()) return; // no temp. dir. created
+
+    if (debug_level_ >= 2)
+    {
+      writeDebug_("Keeping temporary files in directory '" + temp_dir + "'. Set debug level to 1 or lower to remove them.", 2);
+    }
+    else
+    {
+      if (debug_level_ == 1) writeDebug_("Deleting temporary directory '" + temp_dir + "'. Set debug level to 2 or higher to keep it.", 1);
+      File::removeDirRecursively(temp_dir);
+    }
+  }
+  
+  struct LuciphorPSM splitSpecId_(const String& spec_id)
+  {
+    struct LuciphorPSM l_psm;
+    l_psm.spec_id = spec_id;
+    
+    vector<String> parts;
+    spec_id.split(".", parts);
+    l_psm.scan_nr = parts[1].toInt();
+    l_psm.charge = parts[3].toInt();
+    
+    return l_psm;
+  }
+  
+  void getModificationParams_(const ProteinIdentification::SearchParameters& search_params, map<String, String>& modifications)
+  {
+    modifications.clear();
+    boost::regex r(".*\\((.*)\\).*"); // get AA between brackets
+    
```

Comment:
Note that there is e.g. N-term and N-term Q for general and specific N-term mods